linode/instanceip/flatten.go

```diff
@@ -0,0 +1,69 @@
+package instanceip
+
+import (
+	"github.com/linode/linodego"
+)
+
+func flattenIPv4(network *linodego.InstanceIPv4Response) map[string]interface{} {
+	result := make(map[string]interface{})
+
+	result["private"] = flattenIPs(network.Private)
+	result["public"] = flattenIPs(network.Public)
+	result["reserved"] = flattenIPs(network.Reserved)
+	result["shared"] = flattenIPs(network.Shared)
+
+	return result
```

Comment:
Due to SDKv2 limitations this will need to be returned as a list.
e.g.
```suggestion
	return []map[string]any{result}
```